src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -603,25 +603,50 @@ private void RaiseRequestEvent(HttpWebRequest request)
                     activity.Start();
                 }
 
-                request.Headers.Add(RequestIdHeaderName, activity.Id);
-                // we expect baggage to be empty or contain a few items
-                using (IEnumerator<KeyValuePair<string, string>> e = activity.Baggage.GetEnumerator())
+                if (activity.IdFormat == ActivityIdFormat.W3C)
                 {
-                    if (e.MoveNext())
+                    // do not inject header if it was injected already 
+                    // perhaps tracing systems wants to override it
+                    if (request.Headers[TraceParentHeaderName] == null)
```

Comment:
Above the code uses `request.Headers.Get(id)` and here it's using `request.Headers[id]`.  IIRC, one just calls the other, but is there a reason for the different access pattern?

Reply:
no, no particular reason, thanks for noticing it